Content.Client/_White/UI/Emotions/EmotionsUIController.cs

```diff
@@ -0,0 +1,137 @@
+    [Dependency] private readonly IChatManager _chatManager = default!;
+
+
+    private EmotionsWindow? _window;
+    private MenuButton? EmotionsButton => UIManager.GetActiveUIWidgetOrNull<UserInterface.Systems.MenuBar.Widgets.GameTopMenuBar>()?.EmotionsButton;
+
+    private DateTime _lastEmotionTimeUse = DateTime.Now;
+    private const float EmoteCooldown = 1.5f;
+
+    public void OnStateEntered(GameplayState state)
+    {
+        _window = UIManager.CreateWindow<EmotionsWindow>();
+
+        _window.OnOpen += OnWindowOpened;
+        _window.OnClose += OnWindowClosed;
+
+        var emotions = _prototypeManager.EnumeratePrototypes<EmotePrototype>().ToList();
+        emotions.Sort((a,b) => string.Compare(Loc.GetString(a.ButtonText), Loc.GetString(b.ButtonText.ToString()), StringComparison.Ordinal));
+
+        foreach (var emote in emotions)
+        {
+            if (!emote.AllowToEmotionsMenu)
+                continue;
+
+            var button = CreateEmoteButton(emote);
+            _window.EmotionsContainer.AddChild(button);
+        }
+
+        CommandBinds.Builder
+            .Bind(ContentKeyFunctions.OpenEmotionsMenu,
+                InputCmdHandler.FromDelegate(_ => ToggleWindow()))
+            .Register<EmotionsUIController>();
+    }
+
+    public void OnStateExited(GameplayState state)
+    {
+        if (_window != null)
+        {
+            _window.OnOpen -= OnWindowOpened;
+            _window.OnClose -= OnWindowClosed;
+
+            _window.Dispose();
+            _window = null;
+        }
+
+        CommandBinds.Unregister<EmotionsUIController>();
+    }
+
+    public void LoadButton()
+    {
+        if (EmotionsButton == null)
+            return;
+
+        EmotionsButton.OnPressed += EmotionsButtonPressed;
+    }
+
+    public void UnloadButton()
+    {
+        if (EmotionsButton == null)
+            return;
+
+        EmotionsButton.OnPressed -= EmotionsButtonPressed;
+    }
+
+    private void OnWindowOpened()
+    {
+        if (EmotionsButton != null)
+            EmotionsButton.Pressed = true;
+    }
+
+    private void OnWindowClosed()
+    {
+        if (EmotionsButton != null)
+            EmotionsButton.Pressed = false;
+    }
+
+    private void EmotionsButtonPressed(BaseButton.ButtonEventArgs args)
+    {
+        ToggleWindow();
+    }
+
+    private void ToggleWindow()
+    {
+        if (_window == null)
+            return;
+
+        if (_window.IsOpen)
+        {
+            _window.Close();
+            return;
+        }
+
+        _window.Open();
+    }
+
+    private void UseEmote(string emote)
+    {
+        var time = (DateTime.Now - _lastEmotionTimeUse).TotalSeconds;
+        if (time < EmoteCooldown)
+            return;
+
+        _lastEmotionTimeUse = DateTime.Now;
+        _chatManager.SendMessage(emote, ChatSelectChannel.Emotes);
+    }
+
+    private Button CreateEmoteButton(EmotePrototype emote)
+    {
+        var control = new Button();
```

Comment:
Сюда бы еще добавить `ClipText = true;`
И вместо того, чтобы определять параметры потом, лучше их сразу при инициализации определять
```suggestion
        var control = new Button
        {
            ClipText = true,
            HorizontalExpand = true,
            VerticalExpand = true,
            MaxWidth = 250,
            MaxHeight = 45,
            MinWidth = 120,
            TextAlign = Label.AlignMode.Left,
            Text = Loc.GetString(emote.ButtonText)
        };
```